lib/file_transfer_agent/encrypt_util.js

```diff
@@ -125,206 +125,268 @@ function EncryptUtil(encrypt, filestream, temp) {
+   */
+  this.decryptFile = async function (metadata, encryptionMaterial, inputFilePath,
+    tmpDir = null, chunkSize = blockSize * 4 * 1024) {
+    return await this.decryptFileCBC(metadata, encryptionMaterial, inputFilePath, tmpDir, chunkSize);
+  };
+
+  this.decryptFileCBC = async function (metadata, encryptionMaterial, inputFilePath,
     tmpDir = null, chunkSize = blockSize * 4 * 1024) {
-    // Get key and iv from metadata
     const keyBase64 = metadata.key;
     const ivBase64 = metadata.iv;
 
-    // Get decoded key from base64 encoded value
-    const decodedKey = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
-    const keySize = decodedKey.length;
+    const decodedKek = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
 
-    // Get key bytes and iv bytes from base64 encoded value
     const keyBytes = new Buffer.from(keyBase64, BASE64);
     const ivBytes = new Buffer.from(ivBase64, BASE64);
+    const keyDecipher = crypto.createDecipheriv(AES_ECB.cipherName(decodedKek.length), decodedKek, null);
+    const fileKey = performCrypto(keyDecipher, keyBytes);
 
-    // Create temp file
-    let tempOutputFileName;
-    let tempFd;
-    await new Promise((resolve, reject) => {
-      tmp.file({ dir: tmpDir, prefix: path.basename(inFileName) + '#' }, (err, path, fd) => {
+    const dataDecipher = crypto.createDecipheriv(AES_CBC.cipherName(fileKey.length), fileKey, ivBytes);
+    return await performFileStreamCrypto(dataDecipher, tmpDir, inputFilePath, chunkSize);
+  };
+
+  //TODO: Add proper usage when feature is ready (SNOW-940981)
+  this.decryptFileGCM = async function (metadata, encryptionMaterial, inputFilePath, tmpDir = null) {
+    const keyBase64 = metadata.key;
+    const keyIvBase64 = metadata.keyIv;
+    const dataIvBase64 = metadata.iv;
+
+    const decodedKek = Buffer.from(encryptionMaterial[QUERY_STAGE_MASTER_KEY], BASE64);
+    const keyBytes = new Buffer.from(keyBase64, BASE64);
+    const keyIvBytes = new Buffer.from(keyIvBase64, BASE64);
+    const dataIvBytes = new Buffer.from(dataIvBase64, BASE64);
+
+    const fileKey = this.decryptGCM(keyBytes, decodedKek, keyIvBytes, DEFAULT_KEY_AAD);
+
+    const fileContent = await new Promise((resolve, reject) => {
+      fs.readFile(inputFilePath, (err, data) => {
         if (err) {
           reject(err);
         }
-        tempOutputFileName = path;
-        tempFd = fd;
-        resolve();
+        resolve(data);
       });
     });
 
-    // Create key decipher with decoded key and AES ECB
-    let decipher = crypto.createDecipheriv(aesEcb(keySize), decodedKey, null);
-    const fileKey = Buffer.concat([
-      decipher.update(keyBytes),
-      decipher.final()
-    ]);
-
-    // Create decipher with file key, iv bytes, and AES CBC
-    decipher = crypto.createDecipheriv(aesCbc(keySize), fileKey, ivBytes);
+    const decryptedData = this.decryptGCM(fileContent, fileKey, dataIvBytes, DEFAULT_DATA_AAD);
```

Comment:
For decrypting we should use AAD from parameters - it will be stored in file metadata.